src/modules/flow/http-client/http-client.c

```diff
@@ -402,17 +371,102 @@ common_request_finished(void *data,
         "%s Could not parse url contents ", mdata->url);
 }
 
+
+static ssize_t
+sse_received_data_cb(void *data, const struct sol_http_client_connection *conn,
+    struct sol_buffer *buf)
+{
+    struct sol_flow_node *node = data;
+    const size_t prefix_len = strlen("data: ");
+    const size_t suffix_len = strlen("\n\n");
+    const struct http_client_node_type *type;
+    struct sol_str_slice slice;
+    size_t consumed = 0;
+
+    SOL_DBG("Received SSE Data - *%.*s*",
+        SOL_STR_SLICE_PRINT(sol_buffer_get_slice(buf)));
+
+    slice = sol_buffer_get_slice(buf);
+    if (!memmem(slice.data, slice.len, "\n\n", suffix_len))
+        return 0;
+
+    type = (const struct http_client_node_type *)sol_flow_node_get_type(node);
+
+    while (slice.len) {
+        char *start, *end, *content;
+        struct sol_buffer content_buf;
+        size_t content_len, total_len;
+        int r;
+
+        start = memmem(slice.data, slice.len, "data: ", prefix_len);
```

Comment:
it's better to define prefix as a slice before the loop, then `sol_str_slice_starts_with(slice, prefix)` and `sol_str_slice_contains(slice, end)`.